Lib/fontTools/otlLib/builder.py

```diff
@@ -348,6 +348,76 @@ def build(self):
         # buildLookup_
         return self.buildLookup_(subtables)
 
+    def buildFormat2Subtable(self, ruleset, classdefs, chaining=True):
+        st = self.newSubtable_(chaining=chaining)
+        st.Format = 2
+        st.populateDefaults()
+
+        if chaining:
+            (
+                st.BacktrackClassDef,
+                st.InputClassDef,
+                st.LookAheadClassDef,
+            ) = [c.build() for c in classdefs]
+        else:
+            st.ClassDef = classdefs[1].build()
+
+        inClasses = classdefs[1].classes()
+
+        classSets = []
+        for _ in inClasses:
+            classSet = self.newRuleSet_(format=2, chaining=chaining)
+            classSet.populateDefaults()
+            classSets.append(classSet)
+
+        coverage = set()
+        classRuleAttr = self.ruleAttr_(format=2, chaining=chaining)
+
+        for rule in ruleset.rules:
+            rule_table = self.newRule_(format=2, chaining=chaining)
+            if chaining:
+                rule_table.BacktrackGlyphCount = len(rule.prefix)
+                rule_table.LookAheadGlyphCount = len(rule.suffix)
+                rule_table.Backtrack = [
+                    st.BacktrackClassDef.classDefs[list(x)[0]]
```

Comment:
it's unclear to me why you need to cast x to a `list(x)[0]`; what type of values does `rule.prefix` contain? Mind that if x is a set, the order is random so getting the first item may give different results each time you initialize python.

Reply:
I'll add a comment